### PR TITLE
Major enhancement to the docker setup and the config file

### PR DESCRIPTION
**Pull request by TheFrenchGhosty**
Supersede the changes made in #36 

Changes based on https://github.com/PussTheCat-org/docker-priviblur-quay

TL;DR:

- Massively enhance the example config file, dockerfile and the compose file (also make things work OOTB with docker)

Long version:

- config file has been massively enhanced with documentation and now works OOTB with docker (host is now set to "0.0.0.0" by default since it's the standard) - comments "levels" have been standardized with one # for things that can be enabled and 2 for actual comments

- Dockerfile has been redone from basically scratch and optimized in various ways (note: untested with the `COPY` since I clone the whole repo for my image, but you wanted it this way here, so I kept it this way)

- Add some documentation to the compose file (Note the "Taken from" line about redis as been replaced by "Partially taken from" since parts of it was my work, remove `version: "3"` (deprecated since more than a year)

---

Note:

In the config file there's:

```
    # # If you're running Priviblur behind a remote proxy, one or more of the following must be set
    # # can also be set via env variables by captialzing and prefixing with PRIVIBLUR_
    # #
    # # For more information see
    # # https://sanic.dev/en/guide/advanced/proxy-headers.html
    # #
    # # Default: None
    # #
    # forwarded_secret =
    # real_ip_header =
    # proxies_count =
```

Looking at the sanic documentation "must be set" is just plain wrong.

- forwarded_secret
"In order to use the Forwarded header, you should set app.config.FORWARDED_SECRET to a value known to the trusted proxy server. The secret is used to securely identify a specific proxy server."

So this is basically useless?

- real_ip_header

Might only be useful if you want priviblur to know the real IP of the user... which is basically useless?

- proxies_count
"Setting PROXIES_COUNT tells Sanic how deep to look to get an actual IP address for the client."

So it's only useful if you reverse proxy more than once... so useless for most people?

### CODE REVIEW

**Comment by syeopite** [comment]
Should this line be removed then?

https://github.com/syeopite/priviblur/blob/9bf96c49d8db6be3b1be00e05cf9316344c80d30/config.example.toml#L9

I mostly just included it there due to this line in the documentation

> Services behind reverse proxies _must_ configure one or more of the following configuration values:

**Comment by TheFrenchGhosty** [comment]
@syeopite From what I understand it's only if you want to pass the IP to the application (in this case Priviblur)... which seem highly useless in this case.

So I guess I can rewrite this part of the documentation to explain it like that?

**Comment by syeopite** [comment]
Can you just remove that entire section? I think I'll go just go ahead and remove those options entirely within the codebase since Priviblur does not need to know the users IP at all.

**Comment by TheFrenchGhosty** [comment]
@syeopite Sure

**Comment by TheFrenchGhosty** [comment]
@syeopite done

File: config.example.toml [hunk 1]
```diff
@@ -1,50 +1,42 @@
...
-
-# Controls redis cache options
-# Ignore to disable the cache
-#
-# [cache]
+
+# # Controls redis cache options
+# # Ignore to disable the cache
+# # 
+# [cache] # To enable caching Redis is required, uncomment every lines to it
```
Comment (TheFrenchGhosty) [inline config.example.toml:13]: By the way @syeopite I'm not sure "uncomment every lines to it" (I'll fix that typo by that way)  is true, I assumed it is, but maybe if the URL is set the others will just use their default value (written here), am I wrong?

Comment (syeopite) [reply config.example.toml]: Only the url is required to enable caching, the others will just use their default values when unset.

Comment (TheFrenchGhosty) [reply config.example.toml]: Alright, I'll clarify that

Comment (TheFrenchGhosty) [reply config.example.toml]: Done

File: config.example.toml [hunk 2]
```diff
@@ -1,50 +1,43 @@
...
-#
+
+# # Controls redis cache options
+# # Ignore to disable the cache
+# # 
+# # To enable caching, Redis is required, you then have to uncomment "url" and set it accordingly. Every other options in this section will use their default value (indicated next to them) if they are kept commented out.
 # [cache]
+
     # url =
+    # # For docker use the following:
```
Comment (syeopite) [inline config.example.toml:17]: Shouldn't the url for docker be double-commented here?

Comment (TheFrenchGhosty) [reply config.example.toml:17]: it is?

Comment (syeopite) [reply config.example.toml:17]: Oops, I added the review to the wrong line. I meant this

https://github.com/syeopite/priviblur/blob/f032853e55848673f570abbec8b9b149925fb7d4/config.example.toml#L18

Comment (TheFrenchGhosty) [reply config.example.toml:17]: Well, as I said:

>I just made it so users just have to uncomment it instead of the first one
